MatrixSDK/Data/Store/MXMemoryStore/MXMemoryStore.m

```diff
@@ -186,22 +189,22 @@ - (NSArray*)getEventReceipts:(NSString*)roomId eventId:(NSString*)eventId sorted
 
 - (BOOL)storeReceipt:(MXReceiptData*)receipt inRoom:(NSString*)roomId
 {
-    NSMutableDictionary* receiptsByUserId = [receiptsByRoomId objectForKey:roomId];
+    NSMutableDictionary* receiptsByUserId = receiptsByRoomId[roomId];
     
     if (!receiptsByUserId)
     {
         receiptsByUserId = [[NSMutableDictionary alloc] init];
-        [receiptsByRoomId setObject:receiptsByUserId forKey:roomId];
+        receiptsByRoomId[roomId] = receiptsByUserId;
```

Comment:
You have to protect `receiptsByUserId` with a @synchronized as you have done below